src/tape/ensemble.py

```diff
@@ -538,19 +541,38 @@ def calc_nobs(self, by_band=False, label="nobs"):
                 .reset_index()  # break up the multiindex
                 .categorize(columns=[self._band_col])  # retype the band labels as categories
                 .pivot_table(values=self._band_col, index=self._id_col, columns=self._band_col, aggfunc="sum")
-                .repartition(obj_npartitions)  # counts inherits the source partitions
             )  # the pivot_table call makes each band_count a column of the id_col row
 
+            # repartition the result to align with object
+            if self._object.known_divisions:
+                band_counts = band_counts.reset_index().set_index(self._id_col)  # ugly, but need this
```

Comment:
I had to do this to resolve issues with assign not knowing divisions, and I'm really not happy about it. If anyone has ideas on how to more elegantly generate division information, let me know.

Reply:
The grouping is done on the source table, so the source partitions are preserved. And the divisions for the resulting dataframe are not known unless they were known for source. This leads to the problem where the assign call seems to have issues when the divisions for object are known, but the divisions for the new pandas series to assign are not. That being said, I wonder if it'd resolve better if I tried to remove the division information from object, might give that a go

Reply:
The latest commit changes this to just drop divisions from object so it doesn't try to use them as part of the assign